tests/flows/test_vesting.py

```diff
@@ -147,32 +285,462 @@ def test_main(
+        lp_swap_fee_amount=to_wei("0.003"),
+        publish_market_swap_fee_amount=to_wei("0.001"),
+        ss_contract=get_address_of_type(config, "Staking"),
+        base_token_address=ocean_contract.address,
+        base_token_sender=publisher_wallet.address,
+        publisher_address=publisher_wallet.address,
+        publish_market_swap_fee_collector=get_address_of_type(
+            config, "OPFCommunityFeeCollector"
+        ),
+        pool_template_address=get_address_of_type(config, "poolTemplate"),
+        from_wallet=publisher_wallet,
+    )
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    pool_event = erc20_token.get_event_log(
+        ERC721FactoryContract.EVENT_NEW_POOL,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+
+    assert pool_event[0].args.ssContract == side_staking.address
+    bpool = BPool(web3, pool_event[0].args.poolAddress)
+    assert bpool.is_finalized() is True
+
+    # check if vesting is correct
+    side_staking.check_vesting_created_event(
+        block_number=tx_receipt.blockNumber,
+        erc20_token=erc20_token,
+        vested_blocks=vested_blocks,
+        vesting_amount=vesting_amount,
+    )
+    dt_balance_before = erc20_token.balanceOf(erc20_token.get_payment_collector())
+    available_vesting_before = side_staking.get_available_vesting(erc20_token.address)
+    assert available_vesting_before == 0
+
+    # advance checkpoint_block blocks to see if available vesting increased
+    counter = 0
+    for block in range(checkpoint_blocks[-1]):
+        # send dummy transactions to increase block count (not part of the actual functionality)
+        send_ether(publisher_wallet, ZERO_ADDRESS, 1)
+        if block not in checkpoint_blocks:
+            continue
+        available_vesting_after = side_staking.get_available_vesting(
+            erc20_token.address
+        )
         assert (
-            err.value.args[0]
-            == "execution reverted: VM Exception while processing transaction: revert DataTokenTemplate: cap exceeded"
+            available_vesting_after > available_vesting_before
+        ), "Available vesting was not increased!"
+        amount_vested_so_far = side_staking.get_vesting_amount_so_far(
+            erc20_token.address
         )
+        tx_hash = side_staking.get_vesting(erc20_token.address, publisher_wallet)
 
-    assert erc20_token.balanceOf(consumer_wallet.address) == 0
+        # counter is used to count the number of times when get_vesting is called
+        # (it mints another block that change the formula)
+        counter += 1
+        vested_amount = side_staking.get_amount_vested_from_event(
+            tx_hash=tx_hash, erc20_token=erc20_token, from_wallet=publisher_wallet
+        )
+        assert (
+            vested_amount
+            == int(((block + 1 + counter) * vesting_amount) / vested_blocks)
+            - amount_vested_so_far
+        )
```

Comment:
This is super-hard to read. Please reformat so that it's readable. Eg. `target_vesting_amount = int(((block...`, then `assert vested_amount == target_vesting_amount`.